teps/0100-embedded-taskruns-and-runs-status-in-pipelineruns.md

```diff
@@ -240,6 +243,33 @@ While the names of `TaskRuns` and `Runs` are concatenations of the names of the
 too long. Therefore, we include the `PipelineTask` name because tools, such as
 the Tekton Dashboard, would still need the `PipelineTask` name in these situations.
 
+`ConditionChecks` is present in the existing `PipelineRunTaskRunStatus` struct,
+and `WhenExpressions` is present in both the existing `PipelineRunTaskRunStatus`
+and `PipelineRunRunStatus` structs. They provide information which is not available
+from the individual `TaskRun` or `Run` status, since they represent concepts which
+only exist at the `PipelineRun` level. Therefore, they need to be preserved.
+
+To support `ConditionChecks`, we will add a new struct `PipelineRunChildConditionCheckStatus`
+which will hold the names and statuses of condition checks for the `PipelineTask`. It
+will inline the `PipelineRunConditionCheckStatus` currently used in the full embedded
+statuses. This is needed because `PipelineRunConditionCheckStatus` doesn't contain
+the `ConditionCheckName`, which is the equivalent of a `PipelineTask`'s name, just
+`ConditionName`, which is the equivalent of a `TaskRun` or `Run`'s name. Since we're 
+going to store an array of `PipelineRunChildConditionCheckStatus` rather than a map of 
+`ConditionCheckName` to `PipelineRunConditionCheckStatus`, we need the `ConditionCheckName`
+in the new struct.
```

Comment:
given that conditionchecks are taskruns, can we have childreferences to them instead of the whole status, the same way we'll be doing it for taskruns?

Reply:
I don't think we're storing the `ConditionCheckStatus`'s contents anywhere else - `ConditionCheckStatus` is much simpler than `TaskRunStatus`. It's just a `duckv1beta1.Status`, `PodName`, `StartTime`, `CompletionTime`, and an optional `corev1.ContainerState`, vs `TaskRunStatus`'s embedded `RetriesStatus`, `Steps`, `CloudEvents`, `ResourceResults`, `TaskRunResults`, `Sidecars`, and `TaskSpec`. I'll dig to see if we've got the `ConditionCheckStatus` anywhere we can reference to, but if not, I think it's ok to just leave it as is. Particularly since we'll be removing `conditions` in general at some point in the not-too-distant-future.

Reply:
Yeah, verified - `ConditionCheckStatus` is only here. I guess we could reference the actual `TaskRun` behind the scenes, but this seems minimal enough to not be worth the bother to me.

Reply:
good point that we're removing it soon after all and hopefully not many people are using it 👍🏾

Reply:
Aaaah, wait, this can just be a reference to the underlying `ConditionName` and `ConditionCheckName` with the condition check step name from within the `TaskRun`, I think. I forgot that `conditions` aren't separate `TaskRun`s! 

Since this will be stored inside the child reference for the relevant `TaskRun` (meaning it's already got the information needed to look up the `TaskRun`'s status), we already have the `duckv1beta.Status`, `PodName`, `StartTime`, and `CompletionTime` which are used in the `ConditionCheckStatus` available. Then adding the condition check step name to the `ConditionName` and `ConditionCheckName` is enough to get the container state for the condition check step from the `TaskRun`'s status.

Now I just need to figure out if we even need to specify the condition check step name - feels like that probably is one of `ConditionName` or `ConditionCheckName` - I can never remember which of those is which. =)

Reply:
Ok, I've reconsidered - while that's almost certainly a viable option, it's more work than feels worthwhile for something that's not going to be relevant for that much longer anyway, so I am going with the full `ConditionCheckStatus` approach.